### PR TITLE
fix: make lsp hl not be overrode by cursorline

### PR DESCRIPTION
**Pull request by Shane-XB-Qian**
Signed-off-by: shane.xb.qian <shane.qian@foxmail.com>

### CODE REVIEW

File: autoload/lsp/lsp.vim [hunk 1]
```diff
@@ -41,9 +41,9 @@ def LspInitOnce()
 		{name: 'LspDiagHint', text: 'H>', texthl: 'Question',
 						linehl: 'MatchParen'}])
 
-  prop_type_add('LspTextRef', {highlight: 'Search'})
-  prop_type_add('LspReadRef', {highlight: 'DiffChange'})
-  prop_type_add('LspWriteRef', {highlight: 'DiffDelete'})
+  prop_type_add('LspTextRef', {highlight: 'ErrorMsg', override: true})
```
Comment (yegappan) [inline autoload/lsp/lsp.vim:44]: Why is the highlight group changed from 'Search' to 'ErrorMsg'?  The ErrorMsg highlight group uses a red background.  In this case as there is no error, the Search highlight group (with a yellow background) was used.

**Comment by Shane-XB-Qian** [comment]
because the yellow background is very common highlight, you perhaps had manually searched something, or some keyword syntax highlight etc those default is 'yellow'. red background should be more outstanding, besides it should would disappear when winleave/bufleave, so I think it should be better in this case.

--
shane.xb.qian

**Comment by Shane-XB-Qian** [comment]
or if you preferred or insisted the 'yellow', i can change it back to 'search' still, let me know.

**Comment by yegappan** [comment]
Can you update the PR to include only the 'override' change for the prop_type() calls?

File: autoload/lsp/outline.vim [hunk 2]
```diff
@@ -259,7 +259,7 @@ export def OpenOutlineWindow()
     highlight default link LSPTitle Title
   endif
 
-  prop_type_add('LspOutlineHighlight', {bufnr: bufnr(), highlight: 'Search'})
+  prop_type_add('LspOutlineHighlight', {bufnr: bufnr(), highlight: 'diffchange', override: true})
```
Comment (yegappan) [inline autoload/lsp/outline.vim:262]: Can you remove the 'highlight' group change here also?

File: test/unit_tests.vim [hunk 3]
```diff
@@ -523,7 +523,7 @@ enddef
 # Test for :LspGotoDefinition, :LspGotoDeclaration and :LspGotoImpl
 def Test_LspGotoSymbol()
   silent! edit Xtest.cpp
-  sleep 200m
+  sleep 600m
```
Comment (yegappan) [inline test/unit_tests.vim:526]: Is this change needed?

File: autoload/lsp/outline.vim (hunk 2 above)
Comment (Shane-XB-Qian) [reply autoload/lsp/outline.vim:262]: this here probably common choice, e.g tagbar is using this (diffchange) also.
but anyway, i follow if you preferred.

File: test/unit_tests.vim (hunk 3 above)
Comment (Shane-XB-Qian) [reply test/unit_tests.vim:526]: yes, it happened (failed) sometimes, so extend it a bit.
